values/time.go

```diff
@@ -65,6 +72,45 @@ func (t Time) Time() time.Time {
 	return time.Unix(0, int64(t)).UTC()
 }
 
+// Mul will multiply the Duration by a scalar.
+// This multiplies each component of the vector.
+func (d Duration) Mul(scale int) Duration {
+	return d * Duration(scale)
+}
+
+// IsPositive returns true if this is a positive number.
+// It returns false if the number is zero.
+func (d Duration) IsPositive() bool {
+	return d > 0
+}
+
+// IsZero returns true if this is a zero duration.
+func (d Duration) IsZero() bool {
+	return d == 0
+}
+
+// Normalize will normalize the duration to be within the
+// every bound. This is used to take an offset and normalize it
+// to a positive duration that is within the interval duration.
+func (d Duration) Normalize(every Duration) Duration {
+	offset := d
+	if offset < 0 {
+		offset += every * ((offset / -every) + 1)
+	} else if offset > every {
+		offset -= every * (offset / every)
+	}
+	return offset
+}
```

Comment:
I am quite not following this.

For the case `offset > every` isn't it the same as `offset % every`?

Would it be better to write tests for this?

Reply:
Let me remove this one. It was a direct copy/paste from window and doesn't make sense without that refactor.